src/aiida/tools/pytest_fixtures/orm.py

```diff
@@ -0,0 +1,299 @@
+
+    try:
+        yield filepath_private_key
+    finally:
+        filepath_private_key.unlink(missing_ok=True)
+        filepath_public_key.unlink(missing_ok=True)
+
+
+@pytest.fixture
+def aiida_computer(tmp_path) -> t.Callable[[], 'Computer']:
+    """Return a factory to create a new or load an existing :class:`aiida.orm.computers.Computer` instance.
+
+    The database is queried for an existing computer with the same ``label``, ``hostname``, ``scheduler_type`` and
+    ``transport_type``. If it exists, it means it was probably created by this fixture in a previous call and it is
+    simply returned. Otherwise a new instance is created. Note that the computer is not explicitly configured, unless
+    ``configure_kwargs`` are specified.
+
+    The factory has the following signature:
+
+    :param label: The computer label. If not specified, a random UUID4 is used.
+    :param hostname: The hostname of the computer. Defaults to ``localhost``.
+    :param scheduler_type: The scheduler plugin to use. Defaults to ``core.direct``.
+    :param transport_type: The transport plugin to use. Defaults to ``core.local``.
+    :param minimum_job_poll_interval: The default minimum job poll interval to set. Defaults to 0.
+    :param default_mpiprocs_per_machine: The default number of MPI procs to set. Defaults to 1.
+    :param configuration_kwargs: Optional keyword arguments that, if defined, are used to configure the computer
+        by calling :meth:`aiida.orm.computers.Computer.configure`.
+    :return: A stored computer instance.
+    """
+
+    def factory(
+        label: str | None = None,
+        hostname='localhost',
+        scheduler_type='core.direct',
+        transport_type='core.local',
+        minimum_job_poll_interval: int = 0,
+        default_mpiprocs_per_machine: int = 1,
+        configuration_kwargs: dict[t.Any, t.Any] | None = None,
+    ) -> 'Computer':
+        import uuid
+
+        from aiida.common.exceptions import NotExistent
+        from aiida.orm import Computer
+
+        label = label or f'test-computer-{uuid.uuid4().hex}'
+
+        try:
+            computer = Computer.collection.get(
+                label=label, hostname=hostname, scheduler_type=scheduler_type, transport_type=transport_type
+            )
+        except NotExistent:
+            computer = Computer(
+                label=label,
+                hostname=hostname,
+                workdir=str(tmp_path),
+                transport_type=transport_type,
+                scheduler_type=scheduler_type,
+            )
+            computer.store()
+            computer.set_minimum_job_poll_interval(minimum_job_poll_interval)
+            computer.set_default_mpiprocs_per_machine(default_mpiprocs_per_machine)
+
+        if configuration_kwargs:
+            computer.configure(**configuration_kwargs)
+
+        return computer
+
+    return factory
+
+
+@pytest.fixture
+def aiida_computer_local(aiida_computer) -> t.Callable[[], Computer]:
```

Comment:
The `aiida_computer` fixture gives full control in case the user needs it. Most likely it is not used too often though. Really, in most cases the user will just use `aiida_localhost` which is completely done ready to go. The `aiida_computer_local` is there mostly for symmetry with the `aiida_computer_ssh` and in case people just want to customize the label